tools/assetpipelineeditor/camerawidget.cpp

```diff
@@ -257,3 +341,55 @@ void CameraWidget::updateWidgetValues()
         m_ui->bottom->setValue(m_camera ? m_camera->bottom() : 0.0f);
     }
 }
+
+void CameraWidget::read(const QJsonObject &json)
+{
+    if (!m_camera)
+        return;
+
+    m_camera->setPosition(jsonToVector3D(json["position"].toObject()));
+    m_camera->setViewCenter(jsonToVector3D(json["viewCenter"].toObject()));
+    m_camera->setUpVector(jsonToVector3D(json["upVector"].toObject()));
+    m_camera->setNearPlane(static_cast<float>(json["nearPlane"].toDouble()));
+    m_camera->setFarPlane(static_cast<float>(json["farPlane"].toDouble()));
+
+    const auto projectionType = json["projectionType"].toString();
+    if (projectionType == QLatin1String("Perspective")) {
+        m_camera->setProjectionType(Qt3DRender::QCameraLens::PerspectiveProjection);
+        m_camera->setFieldOfView(static_cast<float>(json["fieldOfView"].toDouble()));
+        m_camera->setAspectRatio(static_cast<float>(json["aspectRatio"].toDouble()));
+    } else {
+        m_camera->setProjectionType(Qt3DRender::QCameraLens::OrthographicProjection);
+        m_camera->setLeft(static_cast<float>(json["left"].toDouble()));
+        m_camera->setRight(static_cast<float>(json["right"].toDouble()));
+        m_camera->setTop(static_cast<float>(json["top"].toDouble()));
+        m_camera->setBottom(static_cast<float>(json["bottom"].toDouble()));
+    }
+
+    updateWidgetValues();
+}
+
+void CameraWidget::write(QJsonObject &json)
+{
+    if (!m_camera)
+        return;
+
+    json["position"] = toJson(m_camera->position());
+    json["viewCenter"] = toJson(m_camera->viewCenter());
+    json["upVector"] = toJson(m_camera->upVector());
+    json["nearPlane"] = static_cast<double>(m_camera->nearPlane());
+    json["farPlane"] = static_cast<double>(m_camera->farPlane());
+
+    const bool isPerspective = m_camera->projectionType() == Qt3DRender::QCameraLens::PerspectiveProjection;
+    if (isPerspective) {
+        json["projectionType"] = QLatin1String("Perspective");
+        json["fieldOfView"] = static_cast<double>(m_camera->fieldOfView());
+        json["aspectRatio"] = static_cast<double>(m_camera->aspectRatio());
```

Comment:
is this something we want to save and we can load properly?
it doesn't really hurt but may lead to distorted output after loading in a window with different aspect until aspect changes...

Reply:
Hmm yeah. Let's get it in and see how it behaves - perhaps we can add a follow up to trigger a new fov and aspect ratio or left/right/top/bottom calc to be done somehow.